core/tests/protractor_desktop/publicationAndLibrary.js

```diff
@@ -217,14 +217,23 @@ describe('Permissions for private explorations', function() {
     explorationEditorSettingsTab = explorationEditorPage.getSettingsTab();
     explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
   });
-  it('should be correct for collaborators', function() {
+  it('should not allow adding users if title is not given', function() {
     users.createUser('alice@privileges.com', 'alicePrivileges');
     users.createUser('bob@privileges.com', 'bobPrivileges');
     users.createUser('eve@privileges.com', 'evePrivileges');
 
     users.login('alice@privileges.com');
     workflow.createExploration();
     explorationEditorPage.navigateToSettingsTab();
+
+    workflow.openEditTestRolesForm();
+    expect(workflow.canAddUser()).toBe(false);
```

Comment:
Can you also check whether a proper message is displayed?

Reply:
The message is displayed based on the same condition used to disable the button, so I think it is not needed in this case.

Reply:
E2E tests are also like black box testing, so don't go with the implementation details, check what functionality you expect as an end user.